mapadroid/worker/WorkerQuests.py

```diff
@@ -603,51 +593,38 @@ def _update_injection_settings(self):
         self._mitm_mapper.update_latest(origin=self._origin, key="injected_settings", values_dict=injected_settings)
 
     def _current_position_has_spinnable_stop(self, timestamp: float):
-        latest: dict = self._mitm_mapper.request_latest(self._origin)
-
-        # ensure data from after the given timestamp
-        if latest.get("timestamp_last_data", 0) < timestamp:
-            data_received = self._wait_for_data(timestamp=timestamp, proto_to_wait_for=106)
-            if data_received == LatestReceivedType.UNDEFINED:
-                self._spinnable_data_failure()
-                return False, False
-            latest: dict = self._mitm_mapper.request_latest(self._origin)
-
-        # TODO: can this still happen with the if-clause above?
-        if latest is None or PROTO_NUMBER_FOR_GMO not in latest.keys():
-            self.logger.warning("Can't spin stop - no GMO data available!")
+        type_received, data_received = self._wait_for_data(timestamp=timestamp, proto_to_wait_for=106)
+        if type_received == LatestReceivedType.UNDEFINED:
             self._spinnable_data_failure()
             return False, False
+        latest_proto = data_received.get("payload")
 
-        gmo_cells: list = latest.get(PROTO_NUMBER_FOR_GMO).get("values", {}).get("payload", {}).get("cells",
-                                                                                                    None)
-        pdlocation = latest.get("location", Location(0, 0))
-        pdlocation = Location(0, 0) if not pdlocation else pdlocation
-        stop_found = False
+        gmo_cells: list = latest_proto.get("cells", None)
 
-        if gmo_cells == list():
+        if len(gmo_cells) == 0:
```

Comment:
I think some flake8 modules may complain about this. We should use `if not gmo_cells`